test/streams-test.js

```diff
@@ -331,3 +326,33 @@ describe("Streaming sugar", function() {
     });
 });
 
+describe("Encoding using internal modules with surrogates in separate chunks:", function () {
+    // Generate the expected object of surrogates UT,by input array
+    function generatesrgObj(input) {
```

Comment:
I don't think you can use `generatesrgObj` in any way except by passing its result to `checkEncodeStream`, right? In that case why not move `checkEncodeStream` inside like this?
```
function checkUtf8EncodeStream(input) {
  return checkEncodeStream({
    encoding: "utf8",
    input: input,
    ...
```